hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableSink.java

```diff
@@ -96,9 +96,10 @@ public SinkRuntimeProvider getSinkRuntimeProvider(Context context) {
       pipeline = Pipelines.hoodieStreamWrite(conf, hoodieRecordDataStream);
       // compaction
       if (OptionsResolver.needsAsyncCompaction(conf)) {
-        // use synchronous compaction for bounded source.
+        // use synchronous compaction and clean for bounded source.
         if (context.isBounded()) {
           conf.setBoolean(FlinkOptions.COMPACTION_ASYNC_ENABLED, false);
+          conf.setBoolean(FlinkOptions.CLEAN_ASYNC_ENABLED, false);
         }
```

Comment:
We have another fix in #6515 , does that solve your problem ?

Reply:
Thanks, this PR can solve my problem. This cleaning does not include the commit of the current batch.

Reply:
Yeah, thanks for the explanation, can we try to add a test case here ?

Reply:
ok.

Reply:
The change is missing for some reason ?